lib/matplotlib/colorbar.py

```diff
@@ -942,15 +915,68 @@ def set_alpha(self, alpha):
         """Set the transparency between 0 (transparent) and 1 (opaque)."""
         self.alpha = alpha
 
+    def _set_scale(self, scale, **kwargs):
+        """
+        Set the colorbar long axis scale.
+
+        Parameters
+        ----------
+        value : {"linear", "log", "symlog", "logit", ...} or `.ScaleBase`
+            The axis scale type to apply.
+
+        **kwargs
+            Different keyword arguments are accepted, depending on the scale.
+            See the respective class keyword arguments:
+
+            - `matplotlib.scale.LinearScale`
+            - `matplotlib.scale.LogScale`
+            - `matplotlib.scale.SymmetricalLogScale`
+            - `matplotlib.scale.LogitScale`
+            - `matplotlib.scale.FuncScale`
+
+        Notes
+        -----
+        By default, Matplotlib supports the above mentioned scales.
+        Additionally, custom scales may be registered using
+        `matplotlib.scale.register_scale`. These scales can then also
+        be used here.
+        """
+        if self.orientation == 'vertical':
+            self.ax.set_yscale(scale, **kwargs)
+        else:
+            self.ax.set_xscale(scale, **kwargs)
+        if isinstance(scale, mscale.ScaleBase):
+            self.__scale = scale.name
+        else:
+            self.__scale = scale
+
+    def _get_scale(self):
+        """
+        Return the colorbar's axis scale as a str.
+        """
+        if self.orientation == 'vertical':
+            return self.ax.get_yscale()
+        else:
+            return self.ax.get_xscale()
+
+    def _set_aspect(self, aspect):
```

Comment:
As these two methods appear to be new, private, and never called can we delete them?

Reply:
We _could_ - I think its useful to document that this is what has to happen when we change the aspect, and I _did_ try to make them public, but @greglucas was skeptical.  

I think maybe its less of a big deal to make `set_aspect` public....

Reply:
I'm still skeptical considering you've even avoided using the private versions ;)

Really though, I'm not all that opposed in general, but it is orthogonal to this PR. Maybe cherry-pick the commits over to another branch and add them via a separate PR?

Reply:
OK, `_set_aspect` and `_get_scale` are removed.  Kept `_set_scale` because it is actually useful.